flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -322,7 +317,7 @@ public void start() {
 	 */
 	public CompletableFuture<Acknowledge> start(final JobMasterId newJobMasterId) throws Exception {
 		// make sure we receive RPC and async calls
-		super.start();
+		start();
 
 		return callAsyncWithoutFencing(() -> startJobExecution(newJobMasterId), RpcUtils.INF_TIMEOUT);
```

Comment:
We don't use an `onStart()` here because we need to pass `JobMasterId`?

Reply:
No, because `start` is not being executed in the main thread and we guarantee that `onStart` is always executed by the main thread.

Reply:
```java
callAsyncWithoutFencing(() -> startJobExecution(newJobMasterId), RpcUtils.INF_TIMEOUT);
```

This code is executed in the main thread. If we can access the right JobMasterId, `startJobExecution` can be executed in `onStart`, can't it?

Reply:
Yes, you are right. But the problem is that we don't know the `newJobMasterId`. I plan a follow up PR where I want to remove the `start` and `suspend` logic from the `JobMaster` which will simplify everything. Here is the branch for it: https://github.com/tillrohrmann/flink/tree/removeSuspendFromJobMaster